bingo.js

```diff
@@ -598,13 +670,142 @@ function fillVersionSelection()
 	});
 }
 
-function setSquareColor(square, colorClass)
+function setSquareColor(square, colorClass, isTriggeredByUser = false)
 {
 	ALL_COLOURS.forEach(c => square.removeClass(c));
 	square.addClass(colorClass);
+	if (isTriggeredByUser) {
+		updateCurrentSheetProgress();
+	}
 }
 
-function copySeedToClipboard(id, event)
+function loadBingoProgress(sheetToLoad)
+{
+	/* 
+	 * Expected input for sheetToLoad is [1,3,2,5,..]
+	 * where index of array is numeric value of square id
+	 * and cell value is index of ALL_COLOURS 
+	 */
+	if (typeof sheetToLoad !== 'undefined' && sheetToLoad.length) {
+		forEachSquare((i, square) => {
+			$.each(ALL_COLOURS, function(index, colorClass){
+				square.removeClass(colorClass);
+			});
+			square.addClass(ALL_COLOURS[sheetToLoad[i]]);
+		});
+	}
+}
+
+var sheetProgressDebounce = null;
+function updateCurrentSheetProgress()
+{
+	if(sheetProgressDebounce) clearTimeout(sheetProgressDebounce);
+	sheetProgressDebounce = setTimeout(function() {
+			updateCurrentSheetProgressDebounced()
+	}, 100);
+}
+
+function updateCurrentSheetProgressDebounced()
+{
+	if (! isLocalStorageIsAvailable()) {
+		return;
+	}
+
+	processSheetAndStoreAs(getKeyForCurrentProgress());
+}
+
+function processSheetAndStoreAs(localStorageKey)
+{
+	var squares = generateSquareProgress();
+	var haveMarkedSquares = squares.reduce(function(a,b) { return !!a | !!b; });
+	result = {
+		squares: squares,
+		timestamp: Date.now()
+	};
+	
+	// save only if current sheet have marked squares or already in storage
+	if (haveMarkedSquares || localStorage.getItem(localStorageKey)) {
+		pushNewLocalSetting(localStorageKey, JSON.stringify(result));
+	}
```

Comment:
With [suggested changes about when to call `updateCurrentSheetProgress`](https://github.com/Joshimuz/mcbingo/pull/121/files#r611984340), it seems that these two checks would be unnecessary – `processSheetAndStoreAs` would only be called when we actually do need to save the progress.

```suggestion
	result = {
		squares: squares,
		timestamp: Date.now()
	};
	
	pushNewLocalSetting(localStorageKey, JSON.stringify(result));
```